proposals/0157-mobile-choice-manager.md

```diff
@@ -71,15 +113,35 @@ NS_ASSUME_NONNULL_BEGIN
 
 @protocol SDLChoiceSetDelegate<NSObject>
 
-- (void)choiceSet:(SDLChoiceSet *)choiceSet didSelectChoice:(SDLChoice *)choice withSource:(SDLTriggerSource)source;
+- (void)choiceSet:(SDLChoiceSet *)choiceSet didSelectChoice:(SDLChoiceCell *)choice withSource:(SDLTriggerSource)source;
 - (void)choiceSet:(SDLChoiceSet *)choiceSet didReceiveError:(NSError *)error;
 
-// This is a less common use case, but if the app is using a choice set with search, this is where the text from the search can be retrieved
-- (void)choiceSet:(SDLChoiceSet *)choiceSet didReceiveText:(NSString *)text fromSource:(SDLTriggerSource)source;
+@optional
+// If not implemented, defaults to none
+- (NSArray<SDLChoiceCell *> *)shouldDeleteChoiceCells:(NSArray<SDLChoiceCell *> *)presentedCells;
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+```objc
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol SDLKeyboardDelegate<NSObject>
+
+// This will be sent upon ENTRY_SUBMITTED or ENTRY_VOICE
+- (void)userDidSubmitInput:(NSString *)inputText withSource:(SDLTriggerSource *)source;
+
+// This will be sent if the keyboard event ENTRY_CANCELLED or ENTRY_ABORTED is sent
+- (void)userDidCancelInputWithReason:(SDLKeyboardEvent)event;
 
 @optional
-// If not implemented, defaults to NO
-- (BOOL)shouldDeleteChoiceSet;
+// If any different keyboard properties than exist on the ScreenManager.keyboardConfiguration, then this can be implemented and customized. A SetGlobalProperties will be sent just before the PresentInteraction and the other properties restored after it completes.
```

Comment:
I'm confused by this first sentence.  Should it read "If any different keyboard properties are present than exist..." or "If any different keyboard properties than exist on the ScreenManager.keyboardconfiguration are present..." ?